src/python-common/ceph/deployment/service_spec.py

```diff
@@ -1488,23 +1491,32 @@ def validate(self) -> None:
             raise SpecValidationError('Invalid transport. Valid values are tcp')
 
         if self.log_level:
-            if self.log_level not in ['debug', 'DEBUG',
-                                      'info', 'INFO',
-                                      'warning', 'WARNING',
-                                      'error', 'ERROR',
-                                      'critical', 'CRITICAL']:
+            if self.log_level.lower() not in ['debug',
+                                      'info',
+                                      'warning',
+                                      'error',
+                                      'critical']:
                 raise SpecValidationError(
                     'Invalid log level. Valid values are: debug, info, warning, error, critial')
 
         if self.spdk_log_level:
-            if self.spdk_log_level not in ['debug', 'DEBUG',
-                                           'info', 'INFO',
-                                           'warning', 'WARNING',
-                                           'error', 'ERROR',
-                                           'notice', 'NOTICE']:
+            if self.spdk_log_level.upper() not in ['DEBUG',
+                                           'INFO',
+                                           'WARNING',
+                                           'ERROR',
+                                           'NOTICE']:
```

Comment:
Do these actually need to be upper/lower? Just curious why they're being handled differently.

Reply:
@adk3798 this is because of protobuf annoying handling on enums. You can't have the same name in two different enums. As we have a log level for the gateway and a log level for SPDK with slightly different values but a lot of common values I had to use upper case for one and lower case for the other. Anyway, as I run lower() or upper() on the value I read from the conf file, I don't really care about the case there. So I could have swapped the lower() and upper() in this spec test. Just preferred displaying an error to the user with the right values.

Reply:
It's okay as is. More so just want to know why.